tests/functional/modules/test_zos_zfs_resize_func.py

```diff
@@ -0,0 +1,551 @@
+    trace_destination_file_s = "/" + get_random_file_name(dir="tmp")
+    try:
+        hosts.all.shell(cmd="touch {0}".format(trace_destination_file))
+        mount_folder = set_environment(ansible_zos_module=hosts, ds_name=ds_name)
+
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=grow_size,
+                                            trace_destination=trace_destination_file)
+
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert "grown" in result.get('stdout')
+            assert result.get('new_size') >= grow_size
+            assert result.get('new_size') >= result.get('old_size')
+            assert result.get('new_free_space') >= result.get('old_free_space')
+            assert result.get('space_type') == "K"
+            assert "Printing contents of table at address" in result.get("stdout")
+            cmd = "cat {0}".format(trace_destination_file)
+            output_of_trace_file = hosts.all.shell(cmd=cmd)
+            for out in output_of_trace_file.contacted.values():
+                assert out.get("stdout") is not None
+
+        hosts.all.shell(cmd="touch {0}".format(trace_destination_file_s))
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=shrink_size,
+                                            trace_destination=trace_destination_file_s)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert "shrunk" in result.get('stdout')
+            assert result.get('new_size') <= grow_size
+            assert result.get('new_size') <= result.get('old_size')
+            assert result.get('new_free_space') <= result.get('old_free_space')
+            assert result.get('space_type') == "K"
+            assert "print of in-memory trace table has completed" in result.get('stdout')
+            cmd = "cat {0}".format(trace_destination_file_s)
+            output_of_trace_file = hosts.all.shell(cmd=cmd)
+            for out in output_of_trace_file.contacted.values():
+                assert out.get("stdout") is not None
+
+    finally:
+        clean_up_environment(hosts=hosts, ds_name=ds_name, temp_dir_name=mount_folder)
+
+@pytest.mark.parametrize("trace_destination", ["seq", "pds", "pdse"])
+def test_grow_n_shrink_operations_trace_ds(ansible_zos_module, trace_destination):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    grow_size = 1800
+    shrink_size = 1200
+
+    trace_destination_ds = get_tmp_ds_name()
+    trace_destination_ds_s= get_tmp_ds_name()
+    try:
+        if trace_destination == "seq":
```

Comment:
I think we talked about that the module should create the file and/or trace dataset just like we do with backup_data_sets ? 

Also, Is there a clear error message if the data set provided record length is not enough for the trace output? 
To avoid these issues, is there a possibility we create a ds with record format "U" ?

Reply:
Now I'm making memory I think we talked about users may be wanting to keep their trace files under pds/e such as you do below...